File: 938-range-sum-of-bst/938.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def find(self, root, low, high):
        if root is None:
            return

        if low <= root.val <= high:
            self.sum += root.val

        if root.val > low:
            self.find(root.left, low, high)

        if root.val < high:
            self.find(root.right, low, high)

    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        self.sum = 0
        self.find(root, low, high)
        return self.sum

File: 938-range-sum-of-bst/938.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    int sum;
    void dfs(TreeNode* root, int low, int high) {
        if (!root)
            return;
        if (root->val >= low && root->val <= high)
            sum += root->val;
        if (low <= root->val)
            dfs(root->left, low, high);
        if (high >= root->val)
            dfs(root->right, low, high);
    }
public:
    Solution() : sum(0) { }
    int rangeSumBST(TreeNode* root, int low, int high) {
        dfs(root, low, high);
        return sum;
    }
};
